plugins/leaderboard/lib/api/filters/KalturaUserScorePropertiesFilter.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * @package plugins.leaderboard
+ * @subpackage api.filters
+ * @abstract
+ * @relatedService UserScoreService
+ */
+class KalturaUserScorePropertiesFilter extends KalturaUserScorePropertiesBaseFilter
+{
+	static private $map_between_objects = array
+	(
+		"userIdEqual" => "_eq_puser_id",
+		"userIdIn" => "_in_puser_id",
+		"rankGreaterThanOrEqual" => "_gte_rank",
+		"rankLessThanOrEqual" => "_lte_rank",
+	);
+	
+	static private $order_by_map = array
+	(
+		"+rank" => "+rank",
+		"-rank" => "-rank",
+	);
+	
+	/**
+	 * @var int
+	 */
+	public $virtualEventId;
+	
+	/**
+	 * @var int
+	 */
+	public $leaderboardId;
+	
+	/**
+	 * @var string
+	 */
+	public $userIdEqual;
+	
+	/**
+	 * @var string
+	 */
+	public $userIdIn;
+	
+	/**
+	 * @var int
+	 */
+	public $ranksAboveUser;
+	
+	/**
+	 * @var int
+	 */
+	public $ranksBelowUser;
+	
+	public function getMapBetweenObjects()
+	{
+		return array_merge(parent::getMapBetweenObjects(), self::$map_between_objects);
+	}
+	
+	protected function getCoreFilter()
+	{
+		return null;
+	}
+	
+	private function initRedisWrapper()
+	{
+		$redisWrapper = new kInfraRedisCacheWrapper();
+		$redisConfig = kConf::get('redis', kConfMapNames::RUNTIME_CONFIG);
+		$config = array('host' => $redisConfig['host'], 'port' => $redisConfig['port'], 'timeout' => '0');
+		$redisWrapper->init($config);
+		return $redisWrapper;
+	}
+	
+	private function prepareRedisKey()
+	{
+		$redisKey = kCurrentContext::getCurrentPartnerId();
+		if ($this->virtualEventId)
```

Comment:
we dont enforce virtualEventId anymore so it can be removed from the key
for now, lets enforce only leaderboard id. change it in general to gameObjectId and have also a field for gameObjectType (can be leaderboard, badge, certificate..)